jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/ViewObjectVersionAction.kt

```diff
@@ -3,18 +3,20 @@
 package software.aws.toolkits.jetbrains.services.s3.objectActions
 
 import com.intellij.icons.AllIcons
+import com.intellij.openapi.actionSystem.DataContext
+import software.aws.toolkits.jetbrains.core.utils.getRequiredData
+import software.aws.toolkits.jetbrains.services.s3.editor.S3EditorDataKeys
 import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeNode
 import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeObjectNode
-import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeTable
 import software.aws.toolkits.resources.message
 
-class ViewObjectVersionAction constructor(treeTable: S3TreeTable) :
-    SingleS3ObjectAction(treeTable, message("s3.version.history.view"), AllIcons.Actions.ShowAsTree) {
-
-    override fun performAction(node: S3TreeNode) {
+class ViewObjectVersionAction : SingleS3ObjectAction(message("s3.version.history.view"), AllIcons.Actions.ShowAsTree) {
+    override fun performAction(dataContext: DataContext, node: S3TreeNode) {
         if (node is S3TreeObjectNode) {
             node.showHistory = true
-            treeTable.refresh()
+
+            // TODO: Can we expand the node too
+            dataContext.getRequiredData(S3EditorDataKeys.BUCKET_VIEWER).refresh()
```

Comment:
Doesn't this already expand the node on refresh? Does this mean something different?

Reply:
Nope, we don't expand versions when we load them today. User has to go in and expand the node